core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -664,35 +826,157 @@ void processMeasurementEvent(WriterJob job) {
                             bufPos += Integer.BYTES;
                             long hi = bufPos + 2L * len;
                             job.floatingCharSink.asCharSequence(bufPos, hi);
-                            row.putLong256(colIndex, job.floatingCharSink);
                             bufPos = hi;
+                            final int colType = writer.getMetadata().getColumnType(colIndex);
+                            if (ColumnType.isSymbol(colType)) {
+                                row.putSym(colIndex, job.floatingCharSink);
+                            } else {
+                                throw CairoException.instance(0)
+                                        .put("cast error for line protocol symbol [columnIndex=").put(colIndex)
+                                        .put(", columnType=").put(ColumnType.nameOf(colType))
+                                        .put(']');
+                            }
+                            break;
+                        }
+
+                        case LineTcpParser.ENTITY_TYPE_LONG256: {
+                            int len = Unsafe.getUnsafe().getInt(bufPos);
+                            bufPos += Integer.BYTES;
+                            long hi = bufPos + 2L * len;
+                            job.floatingCharSink.asCharSequence(bufPos, hi);
+                            bufPos = hi;
+                            final int colType = writer.getMetadata().getColumnType(colIndex);
+                            if (ColumnType.isLong256(colType)) {
+                                row.putLong256(colIndex, job.floatingCharSink);
+                            } else {
+                                throw CairoException.instance(0)
+                                        .put("cast error for line protocol long256 [columnIndex=").put(colIndex)
+                                        .put(", columnType=").put(ColumnType.nameOf(colType))
+                                        .put(']');
+                            }
+                            break;
+                        }
+
+                        case LineTcpParser.ENTITY_TYPE_GEOLONG: {
+                            long geoHash = Unsafe.getUnsafe().getLong(bufPos);
+                            bufPos += Long.BYTES;
+                            row.putLong(colIndex, geoHash);
+                            break;
+                        }
+
+                        case LineTcpParser.ENTITY_TYPE_GEOINT: {
+                            int geoHash = Unsafe.getUnsafe().getInt(bufPos);
+                            bufPos += Integer.BYTES;
+                            row.putInt(colIndex, geoHash);
+                            break;
+                        }
+
+                        case LineTcpParser.ENTITY_TYPE_GEOSHORT: {
+                            short geohash = Unsafe.getUnsafe().getShort(bufPos);
+                            bufPos += Short.BYTES;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).